metricbeat/module/kubernetes/controllermanager/controllermanager_test.go

```diff
@@ -0,0 +1,39 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build !integration
+
+package controllermanager
+
+import (
+	"testing"
+
+	"github.com/elastic/beats/metricbeat/helper/prometheus/ptest"
+)
+
+const testFile = "_meta/test/metrics"
+
+func TestEventMapping(t *testing.T) {
+	ptest.TestMetricSet(t, "kubernetes", "controllermanager",
+		ptest.TestCases{
+			{
+				MetricsFile:  "./_meta/test/metrics.controllermanager.1.14",
+				ExpectedFile: "./_meta/test/metrics.controllermanager.1.14.expected",
+			},
+		},
+	)
```

Comment:
I think this is not testing if fields are documented, we may consider changing it to the testdata framework. It would be to replace this with `mbtest.TestDataFiles(t, "kubernetes", "controllermanager")`, create a `_meta/testdata/config.yml` file and move the files to their location in `_meta/testdata`.

Reply:
Please, open an issue for any global improvement.
This is using those _helpers_ and _test libraries_ as is today

Reply:
Ok, I am fine with leaving this with prometheus test helpers, I was thinking that there were some undocumented fields but I was wrong.